Content.Server/StationEvents/GameDirectorSystem.cs

```diff
@@ -0,0 +1,441 @@
+        {
+            if (proto.Abstract)
+                continue;
+
+            if (!proto.HasComponent<GameRuleComponent>(_factory))
+                continue;
+
+            if (!proto.TryGetComponent<StationEventComponent>(out var stationEvent, _factory))
+                continue;
+
+            // Gate here on players, but not on round runtime. The story will probably last long enough for the
+            // event to be ready to run again, we'll check CanRun again before we actually launch the event.
+            if (!_event.CanRun(proto, stationEvent, count.Players, TimeSpan.MaxValue))
+                continue;
+
+            scheduler.PossibleEvents.Add(new PossibleEvent(proto.ID, stationEvent.Chaos));
+        }
+    }
+
+    protected override void Ended(EntityUid uid, GameDirectorComponent scheduler, GameRuleComponent gameRule,
+        GameRuleEndedEvent args)
+    {
+        scheduler.TimeUntilNextEvent = BasicStationEventSchedulerComponent.MinimumTimeUntilFirstEvent;
+    }
+
+    /// <summary>
+    ///   Decide what event to run next
+    /// </summary>
+    protected override void ActiveTick(EntityUid uid, GameDirectorComponent scheduler, GameRuleComponent gameRule, float frameTime)
+    {
+        scheduler.BeatTime += frameTime;
+        if (scheduler.TimeUntilNextEvent > 0)
+        {
+            scheduler.TimeUntilNextEvent -= frameTime;
+            return;
+        }
+
+        // Somewhat expensive process of invoking the metric systems which iterate through entities to determine
+        //   how bad things currently are on the station.
+        ChaosMetrics chaos;
+        var count = CountActivePlayers();
+
+        chaos = _metrics.CalculateChaos();
+        scheduler.CurrChaos = chaos;
+
+        // Decide what story beat to work with (which sets chaos goals)
+        var beat = DetermineNextBeat(scheduler, chaos, count);
+
+        // Pick the best events (which move the station towards the chaos desired by the beat)
+        var bestEvents = ChooseEvents(scheduler, beat, chaos, count);
+
+        // Run the best event here, if we have any to pick from.
+        if (bestEvents.Count > 0)
+        {
+            LogMessage($"Chaos is: {chaos}");
+            // Sorts the possible events and then picks semi-randomly.
+            // when beat.RandomEventLimit is 1 it's always the "best" event picked. Higher values
+            // allow more events to be randomly selected.
+            var chosenEvent = SelectBest(bestEvents, beat.RandomEventLimit);
+
+            _event.RunNamedEvent(chosenEvent.PossibleEvent.PrototypeId);
+
+            // 2 - 6 minutes until the next event is considered.
+            scheduler.TimeUntilNextEvent = _random.NextFloat(120f, 360f);
+        }
+        else
+        {
+            // No events were run. Consider again in 30 seconds.
+            LogMessage($"Chaos is: {chaos} (No events ran)", false);
+            scheduler.TimeUntilNextEvent = 30f;
```

Comment:
Existing code BasicStationEventSchedulerSystem.ResetTimer

I went for more frequent events because the director is not as harsh as the random nature of the extended mode (which is too "slow and boring" currently)

Hmmph this should be fields in the story beat really

```CS
        /// <summary>
        ///   Reset the event timer once the event is done.
        /// </summary>
        private void ResetTimer(BasicStationEventSchedulerComponent component)
        {
            // 5 - 25 minutes. TG does 3-10 but that's pretty frequent
            component.TimeUntilNextEvent = _random.Next(300, 1500);
        }
```